cli/lib/dinstaller_cli/install_config.rb

```diff
@@ -55,6 +57,28 @@ def initialize
       @root = nil
     end
 
+    def dump
+      to_h.to_yaml
+    end
+
+    def to_h
+      {
+        "product"   => product,
+        "languages" => languages,
+        "disks"     => disks,
+        "user"      => {
+          "name"      => user&.name,
+          "fullname"  => user&.fullname,
+          "autologin" => user&.autologin,
+          "password"  => user&.password
+        },
+        "root"      => {
```

Comment:
This couples the top `InstallConfig` class too tightly to its components. I think `InstallConfig::Root` should have its own `to_h` and then here:

```suggestion
        "root"      => root&.to_h || {},
```

and the same for `User` of course